### PR TITLE
Remove smoke test label on failure

### PR DESCRIPTION
**Pull request by MrJnrman**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR fixes the workflow step that removes the run: smoke tests label after smoke tests are done on PRs.

Addresses #31398 .

### How to test the changes in this Pull Request:

1. Create a PR
2. Add run: smoke tests label
3. Wait for tests to complete and observe comments on the associated PR

### CODE REVIEW

File: .github/workflows/pr-smoke-test.yml [hunk 1]
```diff
@@ -96,8 +96,10 @@ jobs:
...
       - name: Remove label from pull request.
-        if: always() && ${{ contains( github.event.pull_request.labels.*.name, format('run{0} smoke tests', ':')) }}
+        if: |
+          always() 
+           && contains( github.event.pull_request.labels.*.name, format('run{0} smoke tests', ':')) 
         uses: actions-ecosystem/action-remove-labels@v1
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          labels: 'run: smoke tests'
+          labels: 'run: smoke tests'
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml]: @MrJnrman Nice work on this. I see from the activity on the PR that this tested great. The only thing that needs to be addressed is the trailing space or tab at the end of this line.

**Review by rrennick — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-smoke-test.yml (hunk 1 above)
Comment (MrJnrman) [reply .github/workflows/pr-smoke-test.yml]: Thanks Ron. The white space has been removed.

**Comment by lanej0** [comment]
Uploading the results is throwing an error about not being able to find `@woocommerce/e2e-environment`.  My Friday brain is preventing me from contributing meaningfully to why this is happening -- but I'm happy to pair on it on Monday to figure out what's going on.

Screen Shot 2022-01-21 at 4 16 51 PM

**Comment by MrJnrman** [comment]
So I used the `resolveLocalE2ePath` method to build the path to the `test-results.json` file. Under the hood, it does this by first resolving the `@woocommerce/e2e-environment` and building a path from there. The issue seems to be that the context in which the `post-results-to-github-pr.js` script is called, it doesn't have access to any of the project modules.

A solution would be to have that script build the path itself. The issue here is that if the `test-results.json` file moves in the future, this script needs to be updated accordingly. It makes me feel a little weird to hard code the path, but given that this script exists solely for the use with our GitHub CI, I believe it should be fine.

**Comment by MrJnrman** [comment]
The script has been updated to find the `test-results.json` file and looks good to go. I've also noticed that the entire test suite is failing against https://wcnightlysmoketest.bluehost-and-a8c.com which doesn't have anything to do with the changes here.

I'll be taking a look into why those tests are failing in a separate PR is necessary.

**Comment by rrennick** [comment]
> The script has been updated to find the test-results.json file and looks good to go.

@MrJnrman I'm going to be writing a PR to move the file output back to its original location. One of the reasons I requested copying the file in https://github.com/woocommerce/woocommerce/pull/31625#issuecomment-1011258539 is that the raw file includes the retries in the summary. Due to that, we will do a small project to replace the file copy with processing the file. In my PR, I'll add a note to the documentation as to its location similar to the screenshots one.

**Comment by MrJnrman** [comment]
Sounds good @rrennick. I'll update this PR once we get yours merged in.

In regards to the failing smoke tests. Its tracked in #31723.

**Comment by rrennick** [comment]
> I'll update this PR once we get yours merged in.

I added the smoke test label to https://github.com/woocommerce/woocommerce/pull/31726. When the run is finished we should know whether the file is found or not.

**Comment by rrennick** [comment]
@MrJnrman The tests haven't finished but the update-woocommerce step found the results file: https://github.com/woocommerce/woocommerce/pull/31726#issuecomment-1021490973

File: packages/js/e2e-environment/bin/post-results-to-github-pr.js [hunk 2]
```diff
@@ -26,7 +34,8 @@ module.exports = async ( { github, context } ) => {
 
 		output = buildOutput( results );
 	} else {
-		output = `## Test Results Not Found!`;
+		output = `## Test Results Not Found! \n\n`;
+		output += 'The path to the `test-results.json` file may need to be updated in the `post-results-to-github-pr.js` script';
```
Comment (rrennick) [inline packages/js/e2e-environment/bin/post-results-to-github-pr.js]: I would end this at `updated.`. Once other teams start using the label they would not be expected to update the script.

**Review by rrennick — CHANGES_REQUESTED** [review]
@MrJnrman The last round of output looks great. This just needs the message tweaked.

File: packages/js/e2e-environment/bin/post-results-to-github-pr.js (hunk 2 above)
Comment (MrJnrman) [reply packages/js/e2e-environment/bin/post-results-to-github-pr.js]: That's a fair point Ron. Message updated!

**Review by rrennick — APPROVED** [review]
LGTM

**Pull request merged**